### ISSUE TITLE
Salt-virt not respecting enable_vnc=True

### ISSUE THREAD

**Issue by dxiri** [saltstack/salt#40212]
### Description of Issue/Question
It seems like sal-virt is ignoring the **enable_vnc=True** option when creating a new VM. 

For example, trying:
salt-run -l debug virt.init centosTEST37 4 2048 salt://centos6-base.img seed=False enable_vnc=true

returns no error, but the resulting XML does NOT contain a vnc graphics or serial adapter, which makes the VM unable to boot.

### Setup
Salt master with 2 minions, one of those minions is the master itself. Both minions are kvm hypervisors with 128GB ram each.

### Steps to Reproduce Issue
1. Have a VM image ready and put it on salt://centos6-base.img
2. salt-run -l debug virt.init centosTESTvm 4 2048 salt://centos6-base.img seed=False enable_vnc=true
3. Check the resulting VM by looking at either the xml with virsh edit centosTEST or virt-manager, you will notice there is no vnc graphics adapter.

**Something that I found while troubleshooting this:**
If I edit templates/virt/libvirt_domain.jinja and remove the conditionals (if statements) surrounding the vnc and the serial console parts, the vnc graphics adapter WORKS when the VM is deployed, that points at: a) some error with the conditionals or b) some error in modules/virt.py when setting those variables. My guess its "b" but haven't found any so far so the error may lie somewhere else 😃 

### Versions Report
Salt Version:
           Salt: 2016.11.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-67-generic
         system: Linux
        version: Ubuntu 16.04 xenial

**Comment by gtmanfred** [comment]
@dxiri is there any chance you could provide me with a `Vagrantfile` that brings up a libvirt/qemu environment that can demonstrate this?

Or something else equivalent.  Unfortunately I we do not have a ton of experience setting up libvirt, and any assistance with that would greatly increase our chance to get this fixed.

Thanks,
Daniel

**Comment by dxiri** [comment]
@gtmanfred I can provide a salt sls file that I used to setup libvirt based on the docs here: 
https://docs.saltstack.com/en/latest/topics/tutorials/cloud_controller.html

For the centos image:
#########centos image generator script#######
#!/bin/bash
OS="--os-variant=rhel6"
Disk="--disk /var/lib/libvirt/images/centos6-base.qcow2,size=8,format=qcow2"
Src="--location=http://mirror.rackspace.com/centos/6/os/x86_64"
Gr="--graphics vnc"
#Gr="--graphics none"
Cpu="--vcpus=2"
Ram="--ram=2048"
Name="centos6-base"
KS_NO_NET="--initrd-inject=/srv/salt/ks.cfg --extra-args ks=file:/ks.cfg --serial pty"
virt-install $OS $Disk $Src $Gr $Cpu $Ram $KS_NO_NET --name=$Name
#########ends centos image generator script#######

After this completes successfully, copy /var/lib/libvirt/images/centos6-base.qcow2 to /srv/salt/centos6-base.img

#########libvirt.sls file########
libvirt-bin:
  pkg.installed: []
  file.line:
    - name: /etc/default/libvirt-bin
    - mode: replace
    - match: 'libvirtd_opts="-d"'
    - content: 'libvirtd_opts="-d -l"'
    - require:
      - pkg: libvirt-bin
  virt.keys:
    - require:
      - pkg: libvirt-bin

/etc/libvirt/libvirtd.conf:
  file.append:
    - text:
      - listen_tls = 0
      - listen_tcp = 1
      - auth_tcp = "none"
      - tcp_port = "16509"

  service.running:
    - name: libvirt-bin

python-libvirt:
  pkg.installed: []

Additional Packages:
  pkg.installed:
    - pkgs:
      - libguestfs-tools
      - virtinst
      - virt-viewer
      - virt-manager
      - virt-what

#Add storage pool to libvirt
salt://scripts/salt-pool.sh:
  cmd.script:
    - unless: virsh pool-info SaltPool

#########end libvirt.sls file########

####saltpool.sh script called above, goes inside 'scripts' folder on the salt root#####
#!/bin/bash
virsh pool-define-as SaltPool --type dir --target /srv/salt-images/
virsh pool-start SaltPool
virsh pool-autostart SaltPool
######ends saltpool.sh script####

So, with those 3 files, you can apply that state to an ubuntu 16.04 LTS server and you would get the exact same setup as I have right now 😄 

Hope it helps!

**Comment by gtmanfred** [comment]
Awesome, thanks!

**Comment by dxiri** [comment]
you are welcome! one tiny change I just noticed, please replace:
- content: 'libvirtd_opts="-d -l"'
with:
- content: 'libvirtd_opts="-l"'

since ubuntu 16 doesn't like the -d option and starting libvirt-bin will fail.

Reference:
https://bugs.launchpad.net/ubuntu/+source/libvirt/+bug/1567272

**Comment by dxiri** [comment]
@gtmanfred has there been any progress on this? is there anything I can do to help?

**Comment by gtmanfred** [comment]
Sorry, I forgot, I have pinned this and will do it tomorrow

**Comment by gtmanfred** [comment]
I need a ks.cfg

Thanks,
Daniel

**Comment by dxiri** [comment]
@gtmanfred Sure! Here:
------- start ks.cfg ------
# Kickstart file automatically generated by anaconda.

#version=DEVEL
install
repo --name="Saltstack" --baseurl=https://repo.saltstack.com/yum/redhat/6/x86_64/latest/
text
reboot
url --url=http://mirror.centos.org/centos/6/os/x86_64
lang en_US.UTF-8
keyboard us
network --onboot yes --device eth0 --bootproto dhcp --ipv6 auto
rootpw  --iscrypted $6$cePymCDK6zjdl9uk$wYHf6wP7PivU2kq4VckjNnnt.b9E08NL5F8UyZ0Q0smJC3CBZc3Z/h4W1Qee5rSZDjpiKsbM8Pys1mBbsHU72/
firewall --service=ssh
authconfig --enableshadow --passalgo=sha512
selinux --enforcing
timezone --utc America/New_York
bootloader --location=mbr --driveorder=vda --append="crashkernel=auto rhgb quiet"
# The following is the partition information you requested
# Note that any partitions you deleted are not expressed
# here so unless you clear all partitions first, this is
# not guaranteed to work
clearpart --all --drives=vda

part /boot --fstype=ext4 --size=500
part pv.253002 --grow --size=1
volgroup vg_centos6base --pesize=4096 pv.253002
logvol / --fstype=ext4 --name=lv_root --vgname=vg_centos6base --grow --size=1024 --maxsize=51200
logvol swap --name=lv_swap --vgname=vg_centos6base --grow --size=819 --maxsize=819

repo --name="CentOS"  --baseurl=http://mirror.centos.org/centos/6/os/x86_64 --cost=100

%packages
@core
@server-policy
@workstation-policy
vim
wget
openssh-clients
salt-minion
%end

-------- ends ks.cfg ----------

Just replace the root password and you are good to go :)

Thank you!

**Comment by gtmanfred** [comment]
I have verified that this is not applying the vnc option.

Unfortunately this is not a priority issue so it will take a while for us to get to it, if you wanted to submit a PR, we would greatly appreciate it.

Thanks for reporting,
Daniel

**Comment by Arabus** [comment]
This has the same root as #43928 - namely the arguments bein passed to the virt module being out of order in the runner.

```salt/modules/virt.py
def init(name,¬
         cpu,¬
         mem,¬
         image=None,¬
         nic='default',¬
         interfaces=None,¬
         hypervisor=None,¬
         start=True,  # pylint: disable=redefined-outer-name¬
         disk='default',¬
         disks=None,¬
         saltenv='base',¬
         seed=True,¬
         install=True,¬
         pub_key=None,¬
         priv_key=None,¬
         seed_cmd='seed.apply',¬
         enable_vnc=False,¬
         enable_qcow=False,¬
         graphics=None,¬
         **kwargs):¬
```
vs.
```salt/runners/virt.py
...
def init(¬
        name,¬
        cpu,¬
        mem,¬
        image,¬
        hypervisor='kvm',¬
        host=None,¬
        seed=True,¬
        nic='default',¬
        install=True,¬
        start=True,¬
        disk='default',¬
        saltenv='base',¬
        enable_vnc=False):¬
...
    try:¬
        cmd_ret = client.cmd_iter(¬
                host,¬
                'virt.init',¬
                [¬
                    name,¬
                    cpu,¬
                    mem,¬
                    image,¬
                    nic,¬
                    hypervisor,¬
                    start,¬
                    disk,¬
                    saltenv,¬
                    seed,¬
                    install,¬
                    pub_key,¬
                    priv_key,¬
                    enable_vnc,¬
                ],¬
                timeout=600)
```

**Comment by rallytime** [comment]
This should be fixed with #48731
